web/robots.py

```diff
@@ -27,9 +24,13 @@ def is_not_empty(self):
 
 def get_robot_parser(url):
     domain = get_domain(url)
+    domain_http_client, headers = select_client(domain)
     if domain not in domain_robot_parsers:
         robot_parser = RobotExclusionRulesParser()
-        robots_txt = web_client.get(urljoin(url, "/robots.txt"))
+        robots_txt = domain_http_client.get(
+            urljoin(url, "/robots.txt"),
+            headers=headers,
```

Comment:
There is a behaviour change here; `headers` are now configured consistently during `robots.txt` access.

That may be better suited to a separate issue and pull request with their own reasoning and potentially test coverage.

Reply:
Filed as #46 and resolved by #47.